### PR TITLE
[SMALLFIX] Create target path for csv sink

### LINKED ISSUE #12702: Automatically create metric directory if metric directory is not created yet
**Is your feature request related to a problem? Please describe.**
The current metric system requires users to create the metrics directory before starting the Alluxio cluster, otherwise, it will keep poping warning message as follows:
```
2021-01-05 13:43:03,305 WARN  CsvReporter - Error writing to Cluster.BytesReadRemoteThroughput
java.io.IOException: No such file or directory
	at java.io.UnixFileSystem.createFileExclusively(Native Method)
	at java.io.File.createNewFile(File.java:1012)
	at com.codahale.metrics.CsvReporter.report(CsvReporter.java:319)
	at com.codahale.metrics.CsvReporter.reportGauge(CsvReporter.java:312)
	at com.codahale.metrics.CsvReporter.report(CsvReporter.java:228)
	at com.codahale.metrics.ScheduledReporter.report(ScheduledReporter.java:253)
	at com.codahale.metrics.ScheduledReporter.lambda$start$0(ScheduledReporter.java:182)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Describe the solution you'd like**
Auto-create metrics directory if it's possible.

**Describe alternatives you've considered**

**Urgency**
Explain why the feature is important

**Additional context**

### PR DESCRIPTION
**Pull request by HelloHorizon**
Fix #12702

### CODE REVIEW

File: core/common/src/main/java/alluxio/metrics/sink/CsvSink.java [hunk 1]
```diff
@@ -28,6 +30,9 @@
  */
 @ThreadSafe
 public class CsvSink implements Sink {
+
```
Comment (apc999) [inline core/common/src/main/java/alluxio/metrics/sink/CsvSink.java:33]: nit: unnecessary empty line

File: core/common/src/main/java/alluxio/metrics/sink/CsvSink.java [hunk 2]
```diff
@@ -28,6 +30,9 @@
...
+  private static final Logger LOG = LoggerFactory.getLogger(CsvSink.class);
+
```
Comment (apc999) [inline core/common/src/main/java/alluxio/metrics/sink/CsvSink.java:35]: ditto

File: core/common/src/main/java/alluxio/metrics/sink/CsvSink.java [hunk 3]
```diff
@@ -38,6 +43,7 @@
 
   private CsvReporter mReporter;
   private Properties mProperties;
+  private File mFile;
```
Comment (apc999) [inline core/common/src/main/java/alluxio/metrics/sink/CsvSink.java:46]: - `final` ? put `final` whenever possible
- should be `mDir` as this indicates poll dir?

File: core/common/src/main/java/alluxio/metrics/sink/CsvSink.java [hunk 4]
```diff
@@ -105,4 +113,19 @@ private TimeUnit getPollUnit() {
...
+   * Create the path for CSV sink if target path is not created.
+   *
+   * @param path CSV target path
+   */
+  private void createPollDir(File path) {
+    if (!path.exists()) {
+      try {
+        path.mkdirs();
+      } catch (SecurityException e) {
+        LOG.warn("Fail to create path {} for CSV sink, {}", path.toString(), e.getMessage());
```
Comment (apc999) [inline core/common/src/main/java/alluxio/metrics/sink/CsvSink.java:127]: nit: `toString()` is implied here

```suggestion
        LOG.warn("Fail to create path {} for CSV sink, {}", path, e.getMessage());
```

**Review by apc999 — CHANGES_REQUESTED** [review]
thanks for the improvement!
minor comments left

**Comment by HelloHorizon** [comment]
Thanks @apc999 for the review! Addressed your comments.

**Review by apc999 — APPROVED** [review]

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**